Library/Source/XMLHttpRequest.cpp

```diff
@@ -2,175 +2,178 @@
-    }
+void XMLHttpRequest::AddEventListener(const Napi::CallbackInfo& info)
+{
+    std::string eventType = info[0].As<Napi::String>().Utf8Value();
+    Napi::Function eventHandler = info[1].As<Napi::Function>();
 
-    void XMLHttpRequest::AddEventListener(const Napi::CallbackInfo& info)
+    const auto& eventHandlerRefs = m_eventHandlerRefs[eventType];
+    for (auto it = eventHandlerRefs.begin(); it != eventHandlerRefs.end(); ++it)
     {
-        std::string eventType = info[0].As<Napi::String>().Utf8Value();
-        Napi::Function eventHandler = info[1].As<Napi::Function>();
-
-        const auto& eventHandlerRefs = m_eventHandlerRefs[eventType];
-        for (auto it = eventHandlerRefs.begin(); it != eventHandlerRefs.end(); ++it)
+        if (it->Value() == eventHandler)
         {
-            if (it->Value() == eventHandler)
-            {
-                throw Napi::Error::New(info.Env(), "Cannot add the same event handler twice");
-            }
+            throw Napi::Error::New(info.Env(), "Cannot add the same event handler twice");
         }
-
-        m_eventHandlerRefs[eventType].push_back(Napi::Persistent(eventHandler));
     }
 
-    void XMLHttpRequest::RemoveEventListener(const Napi::CallbackInfo& info)
+    m_eventHandlerRefs[eventType].push_back(Napi::Persistent(eventHandler));
+}
+
+void XMLHttpRequest::RemoveEventListener(const Napi::CallbackInfo& info)
+{
+    std::string eventType = info[0].As<Napi::String>().Utf8Value();
+    Napi::Function eventHandler = info[1].As<Napi::Function>();
+    auto itType = m_eventHandlerRefs.find(eventType);
+    if (itType != m_eventHandlerRefs.end())
     {
-        std::string eventType = info[0].As<Napi::String>().Utf8Value();
-        Napi::Function eventHandler = info[1].As<Napi::Function>();
-        auto itType = m_eventHandlerRefs.find(eventType);
-        if (itType != m_eventHandlerRefs.end())
+        auto& eventHandlerRefs = itType->second;
+        for (auto it = eventHandlerRefs.begin(); it != eventHandlerRefs.end(); ++it)
         {
-            auto& eventHandlerRefs = itType->second;
-            for (auto it = eventHandlerRefs.begin(); it != eventHandlerRefs.end(); ++it)
+            if (it->Value() == eventHandler)
             {
-                if (it->Value() == eventHandler)
-                {
-                    eventHandlerRefs.erase(it);
-                    break;
-                }
+                eventHandlerRefs.erase(it);
+                break;
             }
         }
     }
+}
+
+void XMLHttpRequest::Open(const Napi::CallbackInfo& info)
+{
+    m_method = info[0].As<Napi::String>().Utf8Value();
+    m_url = m_runtimeImpl.GetAbsoluteUrl(info[1].As<Napi::String>().Utf8Value());
+    SetReadyState(ReadyState::Opened);
+}
 
-    void XMLHttpRequest::Open(const Napi::CallbackInfo& info)
+void XMLHttpRequest::Send(const Napi::CallbackInfo& info)
+{
+    auto lock = m_runtimeImpl.AcquireTaskLock();
+    /* todo
```

Comment:
Another bunch of errors when uncommenting. Definitely the same errors as the previous Arcana one.